### PR TITLE
Clean up FIPS codes and use same method for ZIP codes

### PR DESCRIPTION
**Pull request by zaneselvans**
Add a string cleaning function `pudl.helpers.zero_pad_numeric_string()` which can be
used to standardize columns that are supposed to contain numeric codes of fixed width
stored as strings. These codes are particularly susceptible to corruption through data
type conversions.

This function replaces the similar `zero_pad_zips` function, and can be used to clean up
zip codes and FIPS codes that need to be all numeric and have leading zeroes to be
valid.

The FIPS codes being cleaned up here are the ones associated with coalmines reported in
the `fuel_receipts_costs_eia923` table.

Separately:
* Turn on processing of the eia860m by default.
* Avoid validating the `accumulated_depreciation_ferc1` table, which we're no longer processing. Really this should be removed from the list of working tables.
* Set up automatic pre-commit hooks checking & fixing in CI
* A bunch of little linting issues that the above automatic pre-commit hooks running found.
* Ignore a bunch of 3rd party warnings that come up in the tests. There are more to ignore.
* Updated our EIA ETL debugging notebook to use the new `Settings` objects.

### CODE REVIEW

File: src/pudl/glue/ferc1_eia.py [hunk 1]
```diff
@@ -132,7 +132,8 @@ def get_db_plants_ferc1(
     """
     # Need to be able to use years outside the "valid" range if we're trying
     # to get new plant ID info...
-    unrecognized_years = set(years).difference(DataSource.from_id("ferc1").working_partitions['years'])
+    unrecognized_years = set(years).difference(
+        DataSource.from_id("ferc1").working_partitions['years'])
     if unrecognized_years:
         raise ValueError(
             f"Input years {sorted(unrecognized_years)} not available in FERC 1 DB."
```
Comment (bendnorman) [inline src/pudl/glue/ferc1_eia.py:139]: Can we replace this validation logic with a Ferc1Settings object?

File: src/pudl/helpers.py [hunk 2]
```diff
@@ -1123,33 +1123,59 @@ def cleanstrings_snake(df, cols):
...
+    * Replace any empty strings with NA.
+    * Replace any strings longer than n_digits with NA.
+    * Pad remaining digit-only strings to n_digits length.
+    * Replace (invalid) all-zero codes with NA.
 
     Args:
-        zip_series (pd.Series) : series containing the zipcode values.
-        n_digits(int) : zipcode length (likely 4 or 5 digits).
+        col: The Series to clean. May be numeric, string, object, etc.
+        n_digits: the desired length of the output stringss.
```
Comment (bendnorman) [inline src/pudl/helpers.py:1151]: extra s on `strings`

**Review by bendnorman — CHANGES_REQUESTED** [review]
Looks good. Love the unit tests for the new helper function.

File: src/pudl/helpers.py (hunk 2 above)
Comment (zaneselvans) [reply src/pudl/helpers.py:1151]: Preciousss Stringses 💍

File: src/pudl/glue/ferc1_eia.py (hunk 1 above)
Comment (zaneselvans) [reply src/pudl/glue/ferc1_eia.py]: Like just create a `Ferc1Settings` object using the input `years` and then not use it for anything else? Yeah I guess that's a good idea. Done

**Pull request merged**